### PR TITLE
Adding Predicate supports for Bytes Column without dictionary

### PR DESCRIPTION
**Pull request by fx19880617**
Support predicate for equal/not equ/in/not in/ range

### CODE REVIEW

**Comment by kishoreg** [comment]
- There are cases where request id's can be generated as MD5 which can be represented as byte[]. Searching for a specific request is a common access pattern. 
- Query looks like select * from T where requestId = 'hexEncodedByteString'

Irrespective of the use case, we should support filtering on all data types for sake of completion. Currently, we support predicates on bytes if its dictionary encoded but not if it's stored as RAW.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/InPredicateEvaluatorFactory.java [hunk 1]
```diff
@@ -233,4 +236,24 @@ public boolean applySV(String value) {
       return _matchingValues.contains(value);
     }
   }
+
+  private static final class BytesRawValueBasedInPredicateEvaluator extends BaseRawValueBasedPredicateEvaluator {
+    final Set<String> _matchingValues;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/InPredicateEvaluatorFactory.java]: Storing `Set<ByteArray>` should be faster.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/InPredicateEvaluatorFactory.java [hunk 2]
```diff
@@ -233,4 +236,24 @@ public boolean applySV(String value) {
...
+
+    BytesRawValueBasedInPredicateEvaluator(InPredicate inPredicate) {
+      String[] values = inPredicate.getValues();
+      _matchingValues = new HashSet<>(HashUtil.getMinHashSetSize(values.length));
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/InPredicateEvaluatorFactory.java]: Not related to this pr, but `HashUtil.getMinHashSetSize()` is implemented wrong

File: pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/NotInPredicateEvaluatorFactory.java [hunk 3]
```diff
@@ -251,4 +254,25 @@ public boolean applySV(String value) {
       return !_nonMatchingValues.contains(value);
     }
   }
+
+  private static final class BytesRawValueBasedNotInPredicateEvaluator extends BaseRawValueBasedPredicateEvaluator {
+    final Set<String> _nonMatchingValues;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/NotInPredicateEvaluatorFactory.java]: Use `Set<ByteArray>`

**Pull request merged**